src/clj/matcher_combinators/clj_test.clj

```diff
@@ -207,3 +208,33 @@
                       {clojure.lang.IPersistentMap matchers/equals}
                       msg
                       form))
+
+(defmethod clojure.test/assert-expr 'match-roughly? [msg form]
+  (let [directive (first form)
+        delta     (second form)
+        the-rest  (rest (rest form))
+        y         "sadf"
+        roughly-delta? (fn [expected]
+                         ;; uncomment this line and everything goes to shit
+                         (println y)
```

Comment:
I'm running into a super crazy clojure issue where binding a value outside of this closure and referencing it here leads to 
```
#error {
 :cause "No matching ctor found for class matcher_combinators.clj_test$eval15104$fn__15105$rough
ly_delta_QMARK___15106"
 :via
 [{:type clojure.lang.Compiler$CompilerException
   :message "Syntax error compiling at (matcher_combinators/test_test.clj:95:1)."
   :data #:clojure.error{:phase :compile-syntax-check, :line 95, :column 1, :source "matcher_com
binators/test_test.clj"}
   :at [clojure.lang.Compiler load "Compiler.java" 7648]}
  {:type java.lang.ExceptionInInitializerError
   :message nil
   :at [sun.reflect.NativeConstructorAccessorImpl newInstance0 "NativeConstructorAccessorImpl.ja
va" -2]}
  {:type java.lang.IllegalArgumentException
   :message "No matching ctor found for class matcher_combinators.clj_test$eval15104$fn__15105$r
oughly_delta_QMARK___15106"
   :at [clojure.lang.Reflector invokeConstructor "Reflector.java" 288]}]
 :trace
 [[clojure.lang.Reflector invokeConstructor "Reflector.java" 288]
  [clojure.lang.LispReader$EvalReader invoke "LispReader.java" 1317]
  [clojure.lang.LispReader$DispatchReader invoke "LispReader.java" 853]
  [clojure.lang.LispReader read "LispReader.java" 285]
  [clojure.lang.LispReader read "LispReader.java" 216]
  [clojure.lang.LispReader read "LispReader.java" 205]
  [clojure.lang.RT readString "RT.java" 1878]
  [clojure.lang.RT readString "RT.java" 1873]
  [matcher_combinators.test_test$eval15257 <clinit> "test_test.clj" 95]
  [sun.reflect.NativeConstructorAccessorImpl newInstance0 "NativeConstructorAccessorImpl.java" -
2]
  [sun.reflect.NativeConstructorAccessorImpl newInstance "NativeConstructorAccessorImpl.java" 62
]
  [sun.reflect.DelegatingConstructorAccessorImpl newInstance "DelegatingConstructorAccessorImpl.
java" 45]
  [java.lang.reflect.Constructor newInstance "Constructor.java" 423]
  [java.lang.Class newInstance "Class.java" 442]
  [clojure.lang.Compiler$ObjExpr eval "Compiler.java" 4996]
  [clojure.lang.Compiler eval "Compiler.java" 7176]
  [clojure.lang.Compiler load "Compiler.java" 7636]
  [clojure.lang.RT loadResourceScript "RT.java" 381]
  [clojure.lang.RT loadResourceScript "RT.java" 372]
  [clojure.lang.RT load "RT.java" 459]
  [clojure.lang.RT load "RT.java" 424]
```

this pattern of variable access is needed to implement the actual `roughly-delta` (see commented out code below